index.html

```diff
@@ -5020,180 +5020,154 @@ <h3 id="playback-AudioBufferSourceNode">
-                                  actualLoopEnd</code>, let
-                                  <code>bufferTime</code> equal
-                                  <code>bufferTime - (actualLoopEnd -
-                                  actualLoopStart)</code>.
-                                </li>
-                                <li>While <code>bufferTime</code> &lt;
-                                <code>actualLoopStart</code>, let
-                                <code>bufferTime</code> equal <code>bufferTime
-                                + (actualLoopEnd - actualLoopStart)</code>.
-                                </li>
-                              </ol>
-                            </li>
-                          </ol>
-                        </li>
-                        <li>If <code>bufferTime &gt;= 0</code> and
-                        <code>bufferTime &lt; buffer.duration</code>, emit the
-                        result of <code>playbackSignal(bufferTime)</code> as
-                        the output frame at <code>index</code>.
-                        </li>
-                        <li>Else, emit silence as the output frame at
-                        <code>index</code>.
-                        </li>
-                        <li>Set <code>bufferTime</code> to <code>bufferTime +
-                        (dt * computedPlaybackRate)</code>.
-                        </li>
-                      </ol>
-                    </li>
-                    <li>Increase <code>index</code> by 1.
-                    </li>
-                    <li>Increase <code>currentTime</code> by <code>dt</code>.
-                    </li>
-                  </ol>
-                </li>
-                <li>If <code>currentTime</code> &gt;= <code>stop</code>,
-                consider playback for the node to have ended. No further render
-                quanta for the node will occur.
-                </li>
-              </ol>
-            </li>
-          </ol>
+          <pre>
+// The AudioBuffer and context employed by this node
+var buffer;
+var context;
+
+// The following variables capture attribute and AudioParam values for the node.
+// They are updated k-rate, except for buffer.
+var loop;
+var detune;
+var loopEnd;
+var loopStart;
+var playbackRate;
+
+// State variables for the node's playback
+var start = 0;
+var offset;
+var stop = Infinity;
+
+var bufferTime = 0;
+var started = false;
+var enteredLoop = false;
+var dt;
+
+// Handle invocation of start method call
+function handleStart(when, pos, duration) {
+  if (arguments.length &gt;= 1) {
+    start = when;
+  }
+  offset = pos;
+  if (arguments.length &gt;= 3) {
+    stop = when + duration;
+  }
+}
+
+// Handle invocation of stop method call
+function handleStop(when) {
+  if (arguments.length &gt;= 1) {
+    stop = when;
+  }
+  else {
+    stop = context.currentTime;
+  }
+}
+
+// Interpolate a multi-channel signal value for some sample frame
```

Comment:
"frame" -> "time" seems better because the parameter is a time value, not a frame (index) value. And in the comments you use "frame" to mean the sample values.